bones/exemplar.go

```diff
@@ -4,39 +4,58 @@ import (
 	"fmt"
 	"strings"
 
-	"golang.org/x/exp/slices"
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
 // Formatter is a configurable target language exemplar generator.
 type Formatter struct {
-	Lang       Lang
-	QuoteStyle QuoteStyle
+	lang       Lang
+	quoteStyle QuoteStyle
+	minimal    bool
 
-	messageStack []protoreflect.FullName
+	messagesSeen map[protoreflect.FullName]bool
 }
 
-type FormatOptions = Formatter
+// NewFormatter creates a new Formatter for given language, quote style,
+// and weather to use just input/output type names or flash out all
+// data structures on their first occurrence.
+func NewFormatter(lang Lang, quoteStyle QuoteStyle, minimal bool) *Formatter {
+	return &Formatter{
+		lang:       lang,
+		quoteStyle: quoteStyle,
+		minimal:    minimal,
```

Comment:
messagesSeen not being initialised?

Reply:
initialised in reset, but fair. let's put it in here too.